### PR TITLE
Upgrade puma

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made?

there was a CVE

## How was this change tested?

unit tests so far (integration tests after this is merged)

### CODE REVIEW

**Review by jcoyne — CHANGES_REQUESTED** [review]
can we upgrade to puma 5.3?

File: Gemfile [hunk 1]
```diff
@@ -3,20 +3,6 @@
...
-
-group :development do
-  gem 'dor-services-client' # used by bin/validate-cocina-roundtrip
-  gem 'listen', '~> 3.0.5'
-  gem 'rubyzip', '>= 1.0.0'
-  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
-  gem 'spring'
-  gem 'spring-watcher-listen', '~> 2.0.0'
-end
-
```
Comment (ndushay) [inline Gemfile:19]: having this stuff separate made it harder to find - i put it where it is for most of our apps ...

Comment (jcoyne) [reply Gemfile:19]: I'm seeing this stuff at the top in argo: https://github.com/sul-dlss/argo/blob/main/Gemfile#L6-L9

**Comment by ndushay** [comment]
I'll see if tests pass with puma 5.3, sure.

File: Gemfile (hunk 1 above)
Comment (jcoyne) [reply Gemfile:19]: It is this way in h2 and dor_indexing_app though. 🤷

**Review by jcoyne — APPROVED** [review]

**Pull request merged**